src/io/hdf_archive.h

```diff
@@ -203,6 +235,37 @@ class hdf_archive
     return e.read(p, aname, xfer_plist);
   }
 
+  /* read the data from the group aname and return status
+     use read() for inbuilt error checking
+     @return true if successful
+
+     This is using slicing of the dataspace described in the corresponding 
+     version of read
+  */
+  template<typename T>
+  bool readEntry(T& data, const std::vector<int>& readSpec, const std::string& aname)
+  {
+    if (Mode[NOIO])
+      return true;
+    hid_t p = group_id.empty() ? file_id : group_id.top();
+    h5data_proxy<T> e(data);
+    return e.read(p, aname, readSpec);
+  }
+
+  template<typename T, typename IC>
+  bool readEntry2(T& data, const IC& readSpec, const std::string& aname)
```

Comment:
This is the API used by method 3.
I'm concerned about adding this very specific APIs.
This PR only needs read with a particular pattern. Then if we need to write, add a symmetric API.
Then we need to read/write hyperslab with both offsets and counts, we add another pair of APIs?
Method 2 hyperslab is a more generic method. One for all these cases.